modin/pandas/test/test_dataframe.py

```diff
@@ -304,110 +305,36 @@ def test_math_functions(self, data, function):
-        else:
-            modin_result = getattr(modin_df, op)(modin_df)
-            df_equals(modin_result, pandas_result)
-
-        try:
-            pandas_result = getattr(pandas_df, op)(4)
-        except TypeError:
-            with pytest.raises(TypeError):
-                getattr(modin_df, op)(4)
-        else:
-            modin_result = getattr(modin_df, op)(4)
-            df_equals(modin_result, pandas_result)
-
-        try:
-            pandas_result = getattr(pandas_df, op)(4.0)
-        except TypeError:
-            with pytest.raises(TypeError):
-                getattr(modin_df, op)(4.0)
-        else:
-            modin_result = getattr(modin_df, op)(4.0)
-            df_equals(modin_result, pandas_result)
-
-        try:
-            pandas_result = getattr(pandas_df, op)("a")
-        except TypeError:
-            with pytest.raises(TypeError):
-                repr(getattr(modin_df, op)("a"))
-        else:
-            modin_result = getattr(modin_df, op)("a")
-            df_equals(modin_result, pandas_result)
+    @pytest.mark.parametrize("other", ["as_left", 4, 4.0, "a"])
+    @pytest.mark.parametrize("op", ["eq", "ge", "gt", "le", "lt", "ne"])
+    @pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
+    def test_comparison(self, data, op, other):
+        modin_df = pd.DataFrame(data)
+        pandas_df = pandas.DataFrame(data)
 
-        frame_data = {
-            "{}_other".format(modin_df.columns[0]): [0, 2],
-            modin_df.columns[0]: [0, 19],
-            modin_df.columns[1]: [1, 1],
-        }
-        modin_df2 = pd.DataFrame(frame_data)
-        pandas_df2 = pandas.DataFrame(frame_data)
+        eval_general(
+            modin_df,
+            pandas_df,
+            operation=lambda df, **kwargs: getattr(df, op)(
+                df if other == "as_left" else other
+            ),
+        )
 
-        try:
-            pandas_result = getattr(pandas_df, op)(pandas_df2)
-        except Exception as e:
-            with pytest.raises(type(e)):
-                getattr(modin_df, op)(modin_df2)
-        else:
-            modin_result = getattr(modin_df, op)(modin_df2)
-            df_equals(modin_result, pandas_result)
+    @pytest.mark.parametrize("op", ["eq", "ge", "gt", "le", "lt", "ne"])
+    @pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
+    def test_multi_level_comparison(self, data, op):
+        modin_df_multi_level = pd.DataFrame(data)
 
         new_idx = pandas.MultiIndex.from_tuples(
-            [(i // 4, i // 2, i) for i in modin_df.index]
+            [(i // 4, i // 2, i) for i in modin_df_multi_level.index]
         )
-        modin_df_multi_level = modin_df.copy()
         modin_df_multi_level.index = new_idx
 
         # Defaults to pandas
         with pytest.warns(UserWarning):
             # Operation against self for sanity check
             getattr(modin_df_multi_level, op)(modin_df_multi_level, axis=0, level=1)
```

Comment:
may we rewrite `eval_general` so it would support check for `UserWarnings`, something like that:
```python
with pytest.warns(None) as warning:
    fn(modin_df, **md_kwargs)
    assert (is_default and isinstance(warning.pop(), UserWarning)) or (
        not is_default and len(warning.list) == 0
    )        
```
[pytest about asserting warnings](https://docs.pytest.org/en/3.0.5/recwarn.html#assertwarnings), maybe you'll find more elegant solution

and then just also use `eval_general` in `test_multi_level_comparison`:
```python
    def test_multi_level_comparison(self, data, op):
        new_idx = pandas.MultiIndex.from_tuples(
            [(i // 4, i // 2, i) for i in modin_df_multi_level.index]
        )

        def operation(df):
            df.index = new_idx
            return getattr(df, op)(df, axis=0, level=1)

        eval_general(
            pd.DataFrame(data),
            pandas.DataFrame(data),
            operation=operation,
            is_default=True,
        )
```

Reply:
why not like this:
```python
with pytest.warns(UserWarning):
    eval_general(...)
```

Reply:
Yeap, that's also make sense